File: c++/summaryRanges.cpp
class Solution {
public:

    string entry(int s, int num) {
        if (s == num) {
            return to_string(s);
        }
        return to_string(s) + "->" + to_string(num);
    }

    
    vector<string> summaryRanges(vector<int>& nums) {
        
        vector<string> result;
        
        if (nums.empty()) {
            return result;
        }

        int start = nums[0];
        int prev = nums[0];
        
        for (int i = 1; i <= nums.size(); i++) {
            if (i == nums.size() || nums[i] != prev + 1) {
                result.push_back(entry(start, prev));
                if (i < nums.size()) {
                    start = nums[i];
                }
            }
            if (i < nums.size()) {
                prev = nums[i];
            }
        }
        
        return result;
    }
};

File: python3/summaryRanges.py

iclass Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        
        if len(nums) == 0:
            return []
            
        res = []

        start = nums[0]
        last = nums[0]
        for i in nums[1:]:
            if i == last + 1:
                last = i
                continue

            if start == last:
                res.append(f"{start}")
            else:
                res.append(f"{start}->{last}")

            start = i
            last = i

        if start == last:
            res.append(f"{start}")
        else:
            res.append(f"{start}->{last}")

        return res
